rollout/temlateref.go

```diff
@@ -219,7 +234,23 @@ func (r *informerBasedTemplateResolver) requeueReferencedRollouts(obj interface{
 	if err != nil {
 		return
 	}
+
+	generation := strconv.FormatInt(roMeta.GetGeneration(), 10)
 	for _, ro := range rollouts {
+		un, ok := ro.(*unstructured.Unstructured)
+		if ok {
+			rollout := unstructuredutil.ObjectToRollout(un)
+			updated := annotations.SetRolloutWorkloadRefGeneration(rollout, generation)
+			rollout.Spec.Template.Spec.Containers = []corev1.Container{}
+			if updated {
+				_, err := r.argoprojclientset.ArgoprojV1alpha1().Rollouts(rollout.Namespace).Update(context.TODO(), rollout, v1.UpdateOptions{})
+				if err != nil {
+					log.Errorf("Cannot update the workload-ref/annotation for %s/%s", rollout.GetName(), rollout.GetNamespace())
+				}
+			}
+
+		}
+
 		if key, err := cache.MetaNamespaceKeyFunc(ro); err == nil {
 			r.rolloutWorkQueue.Add(key)
 		}
```

Comment:
There's actually no need to enqueue the rollout to `r.rolloutWorkQueue` anymore with this new approach because the act of annotating the rollout automatically gets the rollout into the workqueue. I think we should remove these three  lines because we may be operating on the rollout prematurely (where the annotation is not reflected in the rollout), which could cause ConflictErrors during reconciliation.

Can we remove these lines?
```
		if key, err := cache.MetaNamespaceKeyFunc(ro); err == nil {
			r.rolloutWorkQueue.Add(key)
		}
```

Reply:
You are correct. Thanks for the explanation.

Reply:
HI, @jessesuen , I tested remove the above 3 lines. However, the unit test failed at

https://github.com/argoproj/argo-rollouts/blob/460262ee353abe60acbae431265a78c5fb86e58b/rollout/temlateref_test.go#L366-L367

The `Get()` does not return any object. Any advice?

Reply:
I think test needs to be modified since we no longer update workqueue directly. So I think this part of test can be removed as well.

Maybe in place of this, we verify the rollout was annotated.

Reply:
True. Will change the test to verify annotation is added.

Reply:
In fact, can we remove the `rolloutWorkQueue` field entirely from NewInformerBasedWorkloadRefResolver?

Reply:
Hi, @jessesuen , I updated the test to verifying the annotations are added for the rollout. However, it looks like the test clientset can't turn `interface{}` to rollout. Could you help to test a look? 

https://github.com/argoproj/argo-rollouts/pull/1198/files#diff-52c4f59609dee33fdeab90f1c17716000fca7922b7524c4e718e114cc1e41afdR243